pkg/hintrunner/zero/zerohint_others.go

```diff
@@ -152,6 +152,204 @@ func createMemEnterScopeHinter(resolver hintReferenceResolver, memset bool) (hin
+			//> else:
+			//>		n_elms = ids.n_elms
+			//>		assert isinstance(n_elms, int) and n_elms >= 0, \
+			//>			f'Invalid value for n_elms. Got: {n_elms}.'
+			//>		if '__find_element_max_size' in globals():
+			//>			assert n_elms <= __find_element_max_size, \
+			//>				f'find_element() can only be used with n_elms<={__find_element_max_size}. ' \
+			//>				f'Got: n_elms={n_elms}.'
+			//>
+			//>		for i in range(n_elms):
+			//>			if memory[array_ptr + elm_size * i] == key:
+			//>				ids.index = i
+			//>				break
+			//>		else:
+			//>			raise ValueError(f'Key {key} was not found.')
+
+			//> array_ptr = ids.array_ptr
+			arrayPtrAddr, err := hinter.ResolveAsAddress(vm, arrayPtr)
+			if err != nil {
+				return err
+			}
+
+			//> elm_size = ids.elm_size
+			elmSizeVal, err := hinter.ResolveAsUint64(vm, elmSize)
+			if err != nil {
+				return err
+			}
+
+			//> assert isinstance(elm_size, int) and elm_size > 0, \
+			//>		f'Invalid value for elm_size. Got: {elm_size}.'
+			if elmSizeVal == 0 {
+				return fmt.Errorf("Invalid value for elm_size. Got: %v", elmSizeVal)
+			}
+
+			//> key = ids.key
+			keyVal, err := hinter.ResolveAsFelt(vm, key)
+			if err != nil {
+				return err
+			}
+
+			//> if '__find_element_index' in globals():
+			//>		ids.index = __find_element_index
+			findElementIndex, err := ctx.ScopeManager.GetVariableValue("__find_element_index")
+			if err == nil {
+				findElementIndex, ok := findElementIndex.(uint64)
+				if !ok {
+					return fmt.Errorf("Invalid value for __find_element_index. Got: %v", findElementIndex)
+				}
+
+				//>		found_key = memory[array_ptr + elm_size * __find_element_index]
+				findElementIndexFelt := new(fp.Element).SetUint64(findElementIndex)
+				findElementIndexMemoryValue := memory.MemoryValueFromFieldElement(findElementIndexFelt)
+				indexValAddr, err := index.GetAddress(vm)
+				if err != nil {
+					return err
+				}
+
+				err = vm.Memory.WriteToAddress(&indexValAddr, &findElementIndexMemoryValue)
+				if err != nil {
+					return err
+				}
+
+				arrayPtrAddr.Offset = arrayPtrAddr.Offset + elmSizeVal*findElementIndex
+				foundKey, err := vm.Memory.ReadFromAddressAsElement(arrayPtrAddr)
+				if err != nil {
+					return err
+				}
+
+				//>		assert found_key == key, \
+				//>			f'Invalid index found in __find_element_index. index: {__find_element_index}, ' \
+				//>			f'expected key {key}, found key: {found_key}.'
+				if foundKey.Cmp(keyVal) != 0 {
+					return fmt.Errorf("Invalid index found in __find_element_index. index: %v, expected key %v, found key: %v", findElementIndex, keyVal, &foundKey)
+				}
+
+				err = ctx.ScopeManager.DeleteVariable("__find_element_index")
+				if err != nil {
+					return err
+				}
+
+			} else {
+				//>		assert isinstance(n_elms, int) and n_elms >= 0, \
+				//>			f'Invalid value for n_elms. Got: {n_elms}.'
+
+				//>		n_elms = ids.n_elms
+				nElms, err := hinter.ResolveAsUint64(vm, nElms)
+				if err != nil {
+					return err
+				}
```

Comment:
why dont you check that `n_elms >= 0` ?

Reply:
bc I resolve it as uint64, which means that `n_elms` will always be positive

Reply:
mmh why are the tests not failing then?

Reply:
why do you say that resolving as uint64 garantees its not zero?

Reply:
My bad, I meant it guarantees it's not negative, which is required by this assertion (`n_elms>=0`)

Reply:
oh damn i was thinking the whole time  n_elms> 0` ...  hum